src/main/java/fr/zcraft/quartzlib/components/i18n/translators/gettext/PluralForms.java

```diff
@@ -0,0 +1,269 @@
+ * sécurité de leurs systèmes et ou de leurs données et, plus généralement,
+ * à l'utiliser et l'exploiter dans les mêmes conditions de sécurité.
+ *
+ * Le fait que vous puissiez accéder à cet en-tête signifie que vous avez
+ * pris connaissance de la licence CeCILL, et que vous en avez accepté les
+ * termes.
+ */
+
+package fr.zcraft.quartzlib.components.i18n.translators.gettext;
+
+import fr.zcraft.quartzlib.tools.PluginLogger;
+import java.util.Locale;
+import java.util.function.Function;
+import javax.script.ScriptEngine;
+import javax.script.ScriptEngineManager;
+import javax.script.ScriptException;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class PluralForms {
+    /**
+     * The default plural forms function when we cannot do anything else. It's the English plural rule.
+     */
+    private static final Function<Long, Integer> FORMS_FUNCTION_FALLBACK = n -> n != 1 ? 1 : 0;
+
+    /**
+     * The number of plural forms for this script.
+     */
+    private final int formsCount;
+
+    /**
+     * The form script, as written in the PO/MO file.
+     */
+    @NotNull
+    private final String formsScript;
+
+    /**
+     * A function to call to compute the correct plural form
+     * from a count.
+     */
+    private final Function<Long, Integer> formsFunction;
+
+    /**
+     * Constructs a new Gettext plural form.
+     *
+     * <p>The script will be matched against a database of known script. If one match, it will be implemented in
+     * pure Java for better performances. Else, we'll fallback on a JavaScript Engine to do the work. If no
+     * engine can be loaded, we'll fallback on hardcoded English rules.
+     *
+     * @param formsCount  The number of plural forms.
+     * @param formsScript The raw script (without “plural=” prefix).
+     */
+    public PluralForms(int formsCount, @Nullable String formsScript) {
+        this.formsCount = formsCount;
+        this.formsScript = formsScript;
+
+        this.formsFunction = computeFormsFunction();
+    }
+
+    public int computePluralForm(long count) {
+        final int index = this.formsFunction.apply(count);
+
+        if (index < 0 || index > formsCount) {
+            return 0;
+        } else {
+            return index;
+        }
+    }
+
+    private Function<Long, Integer> computeFormsFunction() {
+        if (formsScript == null) {
+            return formsFunctionFallback();
+        }
+
+        // We first try if this script is known
+        Function<Long, Integer> function = formsFunctionFromKnownScripts();
+
+        // Else we try two JS engines, and fallback to English rules.
+        if (function == null) {
+            function = formsFunctionFromNashorn();
+        }
+        if (function == null) {
+            function = formsFunctionFromGraalVM();
+        }
+        if (function == null) {
+            function = formsFunctionFallback();
```

Comment:
Maybe usefull to throw a warning when we are in the fallback case. `PluginLogger.warning("Unknown plural rule, fallback on english rule. Please report your langage and the API version to https://github.com/zDevelopers/QuartzLib");`